Makefile

```diff
@@ -188,7 +188,7 @@ CFLAGS_PRE:=\
         -fdata-sections
 
 CFLAGS:=$(FLAGS_CM4) $(CFLAGS_PRE)
-CPPFLAGS:=-fno-exceptions
+CXXFLAGS:=-fno-exceptions
```

Comment:
Yes, these flags were for C++, not the C pre-processor.

Please modify this commit so that all rules that build a C file apply `$(CPPFLAGS) $(CFLAGS)`, and all the rules that build C++ files apply `$(CPPFLAGS) $(CXXFLAGS)`.

The flags will be defined as follows:

```make
CPPFLAGS:=
CXXFLAGS:=-fno-exceptions
```

___


This could probably be done in 2 commits, but I think you would then need to re-shuffle the previous commit that introduces compilation for C++ files. I think it's easier to just modify this one.